### ISSUE TITLE
Relative local file link

### ISSUE THREAD

**Issue by Gaumerie** [marktext/marktext#1688]
### Description

- [x] Can you reproduce the issue? 

### Steps to reproduce

1. create markdown link to relative file e.g.:
  [My File](my_file.pdf)
2. export as a PDF

**Expected behavior:**

  In the PDF, the text link should link to my_file.pdf next to the markdown file location.

**Actual behavior:**

  In the PDF, the text link links to:
 file:///D:/soft/marktext/marktext/resources/app.sar/dist/electron/my_file.pdf

### Versions

- Mark Text: v0.15.1 and v0.16.0
- Operating system: Windows 10

**Comment by fxha** [comment]
@Jocs I would categorized the issue as bug?!

**Comment by Jocs** [comment]
> I would categorized the issue as bug?!

Yes, it's also a bug.

**Comment by MrHeer** [comment]
I want to fix it, but I checked `editor.vue`, `printService.js`, `editor.js` and `file.js` not find the reason, who can help me?

**Comment by Jocs** [comment]
I am not sure why this link because `file:///D:/soft/marktext/marktext/resources/app.sar/dist/electron/my_file.pdf` when export PDF, so I think we can solve this issue from here.

Yes, we use `win.webContents.printToPDF` to export PDF file.

**Comment by MrHeer** [comment]
When I test the `./Untitled-2.md` will become to `http://localhost:9091/Untitled-2.md`

**Comment by fxha** [comment]
@MrHeer  @Jocs  `http://localhost:9091/` and `file:///D:/soft/marktext/marktext/resources/app.sar/dist/electron` are the absolute paths at runtime. We or the printer used by Electron seem to combine the paths to an absolute path. There is already a workaround for images [here](https://github.com/marktext/marktext/blob/c90bdadb90fd1e5d0d49252424e6ebb663c56f95/src/renderer/services/printService.js#L17).

**Comment by MrHeer** [comment]
I think they are different.
1. images will resolve relative path to absolute path
2. the path of links after exporting pdf will automatically become the absolute path

**Comment by fxha** [comment]
@MrHeer I think this is an issue with marked.js, [here](https://github.com/marktext/marktext/blob/866e7cf53d7e812e44d536fbee7383775421a6da/src/muya/lib/parser/marked/utils.js#L66). From markedjs point of view, converting a relative path to an absolute makes sense because by moving the file an unauthorized user may read files from a server without permissions.

**Comment by MrHeer** [comment]
Yes, converting a relative path to an absolute makes sense, but I checked that the `html` data while export pdf. I found the url is relative path, but it will automatically become the absolute path after exporting pdf.

I did the following experiment：

1. create a `test.html` file, and edit it
	```html
	<a href="./test.txt">go to test.txt</a>
	```
2. create a `test.txt` file in the same directory, and edit it
	```txt
	test.txt
	```
3. open `test.html` with `Chrome` and click `go to test.txt` will will jump to `test.txt`
4. Export `test.html` as pdf in `Chrome` using `Print`, then you can found the relative path will automatically become the absolute path.

So, I think the reason is `Chrome Printer`.

**Comment by ffernn-dev** [comment]
Any update on this? It's blocking for me at the moment as I rely on wiki-style links for my notes
